web/auth/auth_test.go

```diff
@@ -2,425 +2,289 @@ package auth_test
 
 import (
 	"net/http"
-	"net/http/httptest"
-	"reflect"
 	"testing"
 
-	"github.com/gorilla/sessions"
-	"github.com/labstack/echo-contrib/session"
-	"github.com/labstack/echo/v4"
 	"github.com/markbates/goth"
-	"github.com/markbates/goth/gothic"
+	"github.com/quickfeed/quickfeed/database"
 	"github.com/quickfeed/quickfeed/internal/qtest"
-	"github.com/quickfeed/quickfeed/qf"
 	"github.com/quickfeed/quickfeed/qlog"
 	"github.com/quickfeed/quickfeed/web/auth"
+	"github.com/steinfletcher/apitest"
+	"gotest.tools/assert"
 )
 
 const (
-	loginRedirect  = "/login"
-	logoutRedirect = "/logout"
-
-	authURL   = "/auth?provider=fake&redirect=" + loginRedirect
-	logoutURL = "/logout?provider=fake&redirect=" + logoutRedirect
-
-	fakeSessionKey  = "fake"
-	fakeSessionName = fakeSessionKey + gothic.SessionName
+	testSecret = "top-secret"
 )
 
 func init() {
 	goth.UseProviders(&auth.FakeProvider{
-		Callback: "/auth/fake/callback",
+		Callback: "/auth/callback/fake",
```

Comment:
Would it make sense to add something like this to the `web` package, e.g., in `webserver.go`:
```go
const (
	Auth     = "/auth/"
	Teacher  = "/auth/teacher/"
	Callback = "/auth/callback/"
	Logout   = "/logout"
	Hook     = "/hook/"
)
```
And use them here, e.g.:
```go
Callback: web.Callback + "fake",
```
And similar for others below??

(Just in case we want to change the endpoints later...)